### PR TITLE
Approval queue locking

### PR DESCRIPTION
**Pull request by Ewoutvans**
This PR introduces the concept of 'reviews' this to facilitate #329 number 1 & 2

Demonstration:
**New video: https://youtu.be/GvdAqN38q7o**
Old Video: https://youtu.be/z4WvOEyP2i4
Old video: https://youtu.be/6Rzh4gy-_Wo

New queue:
queue

### CODE REVIEW

File: app/models/admin/Review.scala [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import db.Model
+import db.impl.ReviewTable
+import db.impl.model.OreModel
+import db.impl.schema.ReviewSchema
+import db.impl.table.ModelKeys._
+import models.project.{Project, Version, Page}
+
+
+/**
+  * Represents a approval instance of [[Project]] [[Version]].
```
Comment (phase) [inline app/models/admin/Review.scala]: `an approval`

File: app/controllers/Application.scala [hunk 2]
```diff
@@ -34,6 +35,8 @@ final class Application @Inject()(data: DataHelper,
 
   private def FlagAction = Authenticated andThen PermissionAction[AuthRequest](ReviewFlags)
 
+
```
Comment (phase) [inline app/controllers/Application.scala]: Extra whitespace?

File: app/controllers/Reviews.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        }
+      }
+    }
+  }
+
+  def takeoverReview(author: String, slug: String, versionString: String) = {
+    (Authenticated andThen PermissionAction[AuthRequest](ReviewProjects)) { implicit request =>
+      withProject(author, slug) { implicit project =>
+        withVersion(versionString) { version =>
+          // Close old  review
```
Comment (phase) [inline app/controllers/Reviews.scala]: Whitespace

File: app/views/projects/versions/view.scala.html [hunk 4]
```diff
@@ -78,7 +78,7 @@ <h1 class="pull-left">@version.versionString</h1>
                     <div class="version-buttons pull-right">
                         <div><span class="date">@version.humanFileSize</span></div>
 
-                        <div class="btn-group">
+                        <div> <!--class="btn-group"-->
```
Comment (phase) [inline app/views/projects/versions/view.scala.html]: Remove this comment

File: conf/application.conf.template [hunk 5]
```diff
@@ -67,6 +67,7 @@ security {
 # Ore configuration
 ore {
     date-format =  "MMM dd, yyyy"
+    date-and-time-format = "yyyy-MM-dd HH:mm"
```
Comment (phase) [inline conf/application.conf.template]: The date format should be the same as the above line.

**Review by phase — CHANGES_REQUESTED** [review]
This is looking awesome! I skimmed the code a little bit. Is there any pagnation in place for the approval queue? That's the big problem the current queue has.

**Comment by Ewoutvans** [comment]
@phase I'm doing pagination in a separate PR once this get merged. @progwml6 asked us to finish this one and start working on other issues

**Comment by felixoi** [comment]
Intentional question: How to reject a version?

**Comment by Ewoutvans** [comment]
Rejecting isn't part of this PR (this is queue locking)

File: conf/application.conf.template (hunk 5 above)
Comment (felixoi) [reply conf/application.conf.template]: I don't think so. Sometimes you don't need time information (hours, minutes, seconds...), that's why I think it's fine to have a date format and a date format with time information.

Comment (phase) [reply conf/application.conf.template]: But they should be consistent, like:
```
    date-format =  "MMM dd, yyyy"
    date-and-time-format = "MMM dd, yyyy HH:mm"
```

Comment (felixoi) [reply conf/application.conf.template]: Oh yeah, I agree.

**Review by phase — APPROVED** [review]

**Review by felixoi — APPROVED** [review]

**Pull request merged**